packages/babel-compat-data/scripts/build-data.js

```diff
@@ -33,8 +33,9 @@ for (const target of ["plugin", "corejs2-built-in"]) {
     newData["transform-export-namespace-from"] = exportNamespaceFromCompatData;
 
     // Add proposal-* aliases for backward compatibility.
-    // TODO(Babel 7): This won't be needed anymore.
-    newData = defineLegacyPluginAliases(newData);
+    if (!process.env.BABEL_8_BREAKING) {
+      newData = defineLegacyPluginAliases(newData);
+    }
```

Comment:
I dislike how this causes us to ship both Babel 7 and Babel 8 data to our users, as well as we then need to keep the flag in the CJS proxies (like `packages/babel-compat-data/plugin-bugfixes.js`)

Reply:
I agree. I think it is fine to keep `-next.json` in the git repo but we can skip them during publish. As for Babel 8 resolving, maybe we can add yarn condition on the `exports` of `@babel/compa-data` and redirect `./plugins.json` to `./plugins-next.json`.